internal/access/access_key.go

```diff
@@ -89,12 +87,15 @@ func DeleteAccessKey(c *gin.Context, id uid.ID, name string) error {
 	if key.IssuedFor == rCtx.Authenticated.User.ID {
 		// users can delete their own keys
 	} else {
-		_, err := RequireInfraRole(c, models.InfraAdminRole)
-		if err != nil {
+		if err := IsAuthorized(rCtx, models.InfraAdminRole); err != nil {
 			return HandleAuthErr(err, "access key", "delete", models.InfraAdminRole)
 		}
 	}
 
+	if rCtx.Authenticated.AccessKey.ID == key.ID {
+		return fmt.Errorf("%w: cannot delete the access key used by this request", internal.ErrBadRequest)
+	}
```

Comment:
This is the primary change to behaviour.  If anyone has suggestions to improve the wording I'm happy to make those changes.